src/main/java/com/o19s/es/ltr/stats/suppliers/StoreStatsSupplier.java

```diff
@@ -0,0 +1,153 @@
+
+    public enum Stat {
+        STORE_STATUS("status"),
+        STORE_FEATURE_COUNT("feature_count"),
+        STORE_FEATURE_SET_COUNT("featureset_count"),
+        STORE_MODEL_COUNT("model_count");
+
+        private final String name;
+
+        Stat(String name) {
+            this.name = name;
+        }
+
+        public String getName() {
+            return name;
+        }
+    }
+
+    public StoreStatsSupplier(Client client, ClusterService clusterService) {
+        this.client = client;
+        this.clusterService = clusterService;
+        this.indexNameExpressionResolver = new IndexNameExpressionResolver();
+    }
+
+    @Override
+    public Map<String, Map<String, Object>> get() {
+        String[] names = indexNameExpressionResolver.concreteIndexNames(clusterService.state(),
+                new ClusterStateRequest().indices(
+                        IndexFeatureStore.DEFAULT_STORE, IndexFeatureStore.STORE_PREFIX + "*"));
+        final MultiSearchRequestBuilder requestBuilder = client.prepareMultiSearch();
+        List<String> indices = new ArrayList<>();
+        Stream.of(names)
+                .filter(IndexFeatureStore::isIndexStore)
+                .map(s -> clusterService.state().metadata().index(s))
+                .filter(Objects::nonNull)
+                .map(IndexMetadata::getIndex)
+                .map(Index::getName)
+                .forEach(idx -> {
+                    indices.add(idx);
+                    requestBuilder.add(countSearchRequest(idx));
+                });
+        return createStoreStatsResponse(requestBuilder, indices);
+    }
+
+    private Map<String, Map<String, Object>> createStoreStatsResponse(MultiSearchRequestBuilder requestBuilder,
+                                                                      List<String> indices) {
+        try {
+            MultiSearchResponse msr = requestBuilder.execute().get();
+            assert indices.size() == msr.getResponses().length;
+            Map<String, Map<String, Object>> stats = new HashMap<>(indices.size());
+
+            Iterator<String> indicesItr = indices.iterator();
+            Iterator<MultiSearchResponse.Item> responseItr = msr.iterator();
+            while (indicesItr.hasNext() && responseItr.hasNext()) {
+                MultiSearchResponse.Item it = responseItr.next();
+                String index = indicesItr.next();
+                Map<String, Object> storeStat = initStoreStat(index);
+                stats.put(IndexFeatureStore.storeName(index), storeStat);
+                if (!it.isFailure()) {
+                    Terms aggs = it.getResponse()
+                            .getAggregations()
+                            .get(AGG_FIELD);
+                    aggs.getBuckets()
+                            .stream()
+                            .filter(Objects::nonNull)
+                            .forEach(bucket -> updateCount(bucket, storeStat));
```

Comment:
sorry my bad, I misread the scope of storeStat in that case I think it'd make more sense to collect and then merge, e.g.:
```
                    aggs.getBuckets()
                            .stream()
                            .filter(Objects::nonNull)
                            .collect(Collectors.toMap(Terms.Bucket::getKeyAsString, Terms.Bucket::getDocCount))
                            .forEach(storeStat::put);
```